### PR TITLE
release(2020-07-07): bump package versions

### PR DESCRIPTION
**Pull request by timtay-microsoft**
old
{
    "device": "1.23.0",
    "service": "1.23.0",
    "deps": "0.9.5",
    "securityProvider": "1.3.0",
    "tpmEmulator": "1.1.1",
    "tpmHsm": "1.1.2",
    "diceEmulator": "1.1.1",
    "dice": "1.1.1",
    "x509": "1.1.4",
    "provisioningDevice": "1.8.2",
    "provisioningService": "1.6.1"
}

new
{
    "device": "1.24.0",
    "service": "1.24.0",
    "deps": "0.10.0",
    "securityProvider": "1.3.0",
    "tpmEmulator": "1.1.1",
    "tpmHsm": "1.1.2",
    "diceEmulator": "1.1.1",
    "dice": "1.1.1",
    "x509": "1.1.4",
    "provisioningDevice": "1.8.3",
    "provisioningService": "1.6.2"
}

## Java IotHub Service Client (com.microsoft.azure.sdk.iot:iot-service-client:1.24.0)

- Add support for setting HTTP read and connect timeouts for registry manager operations (#815)

## Java IotHub Device Client (com.microsoft.azure.sdk.iot:iot-device-client:1.24.0)

- Add new File upload APIs for more granular control over the now deprecated file upload API (#808)
- Add PnP ModelId option to ModuleClient constructors
- Deprecated Client constructors that take x509 authentication through strings/file paths in favor of SSLContext
- Add message type (twin subscribe message, for instance) to log statements
- Cut down on needless spawning of send/receive threads. Now threads stop spawning until work becomes available (#826)

### Bug fixes
- Fix AMQP layer being too proactive when renewing sas tokens (#793)
- Fix AMQP layer not maintaining separate sas token renewal schedules per multiplexed client (#793)
- Fix bug where client library spawned send/receive threads even when in a reconnecting state (#826)

## Java Iot Deps (com.microsoft.azure.sdk.iot:iot-deps:1.10.0)

- Add new File upload APIs for more granular control over the now deprecated file upload API (#808)

### Bug fixes
- Fix issue where HTTP requests didn't send content-length header (#833)

### Bug fixes
- Fix issue where HTTP requests didn't send content-length header (#833)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,11 +33,11 @@
...
         <x509-provider-artifact-id>x509-provider</x509-provider-artifact-id>
 
-        <iot-device-client-version>1.23.0</iot-device-client-version>
-        <iot-service-client-version>1.23.0</iot-service-client-version>
-        <iot-deps-version>0.9.5</iot-deps-version>
-        <provisioning-device-client-version>1.8.2</provisioning-device-client-version>
-        <provisioning-service-client-version>1.6.1</provisioning-service-client-version>
+        <iot-device-client-version>1.24.0</iot-device-client-version>
+        <iot-service-client-version>1.24.0</iot-service-client-version>
+        <iot-deps-version>0.9.6</iot-deps-version>
```
Comment (abhipsaMisra) [inline pom.xml:38]: the PR description says that this should be `0.10.0`?

Comment (timtay-microsoft) [reply pom.xml:38]: Good catch, I missed that one

Comment (timtay-microsoft) [reply pom.xml]: Now it's fixed

**Comment by abhipsaMisra** [comment]
@timtay-microsoft Isn't the version file maintained in the "internals" repo, and updated by the builder?

**Comment by timtay-microsoft** [comment]
Yes, right here: https://github.com/Azure/iot-sdks-internals/blob/master/release/java/master/versions.json

**Comment by timtay-microsoft** [comment]
I originally forgot to bump the deps version in that location, hence why it wasn't bumped here either. I manually fixed that mistake in both repos

**Pull request merged**